extensions/hibernate-orm/deployment/src/main/java/io/quarkus/hibernate/orm/deployment/HibernateOrmConfig.java

```diff
@@ -250,8 +253,8 @@ public boolean isAnyPropertySet() {
         /**
          * The charset of the database.
          */
-        @ConfigItem
-        public Optional<String> charset;
+        @ConfigItem(defaultValue = "UTF-8")
+        public Optional<Charset> charset;
```

Comment:
So let's not make it an Optional anymore.

Reply:
Good point, done